packages/ifpack2/src/Ifpack2_Details_OneLevelFactory_def.hpp

```diff
@@ -224,6 +224,113 @@ OneLevelFactory<MatrixType>::create (const std::string& precType,
   return prec;
 }
 
+template<class MatrixType>
+bool
+OneLevelFactory<MatrixType>::isSupported (const std::string& precType) const
+{
+  // precTypeUpper is the upper-case version of precType.
+  std::string precTypeUpper (precType);
+  if (precTypeUpper.size () > 0) {
+    std::locale locale;
+    for (size_t k = 0; k < precTypeUpper.size (); ++k) {
+      precTypeUpper[k] = std::toupper<char> (precTypeUpper[k], locale);
+    }
+  }
+
+  if (precTypeUpper == "CHEBYSHEV") {
```

Comment:
Why not just define an array of the permitted names, like this?
```cpp
const char permittedNames[] = {
  "CHEBYSHEV", "DENSE", "LAPACK", "AMESOS2", /* ... etc. ... */ };
const size_t numPermittedNames = /* whatever it is */;
const auto end = permittedNames + numPermittedNames;
auto it = std::find(precTypeUpper, permittedNames, end);
return it != end;
```
Just because it was there before doesn't mean it's the best way to write the code (even if I wrote it before ;-) ).